include/device.h

```diff
@@ -613,6 +613,29 @@ static inline bool device_is_ready(const struct device *dev)
 	return device_usable_check(dev) == 0;
 }
 
+/** @brief Check whether a device supports PM.
+ *
+ * This checks whether a device supports PM, returning 0 if it can, and
+ * distinct error values that identify the reason if it cannot.
+ *
+ * @param dev pointer to the device in question.
+ *
+ * @retval 0 if the device is usable.
+ * @retval -ENOSYS if the device has not PM implementeted.
+ * @retval -ENOTSUP if the target was built without device pm enabled.
+ */
+static inline int device_has_pm(const struct device *dev)
+{
+#ifdef CONFIG_PM_DEVICE
+	if (dev->pm_control == NULL) {
+		return -ENOSYS;
+	} else {
+		return 0;
+	}
+#endif
+	return -ENOTSUP;
```

Comment:
I'm not convinced about this API:

1. We're adding a PM related API to `device.h`
2. It should return a boolean, so it can be simplified by just `return dev->pm_control != NULL`. Which makes me think if it's really needed.

Reply:
The information is needed. The function itself is other history, this is a helper that we may change in the future if the internal pm implementation changes. Right now, the device_runtime can just check if a device implements pm_control. Having a function seems cleaner. This can be moved to include/pm/device.h indeed.